src/main/scala/is/hail/variant/GenotypeStream.scala

```diff
@@ -12,13 +12,17 @@ import is.hail.utils.ByteIterator
 
 import scala.collection.mutable
 
-// FIXME use zipWithIndex
 class GenotypeStreamIterator(nAlleles: Int, isDosage: Boolean, b: ByteIterator) extends Iterator[Genotype] {
   override def hasNext: Boolean = b.hasNext
 
-  override def next(): Genotype = {
-    Genotype.read(nAlleles, isDosage, b)
-  }
+  override def next(): Genotype = Genotype.read(nAlleles, isDosage, b)
+}
+
+// FIXME: Implement unboxed Int iterator
```

Comment:
You want to do this?  I'm in favor.  I think it should look like this:

```
abstract class IntIterator extends Iterator[Int] {
  def nextInt(): Int
  override def next(): Int = nextInt()
}
```

and `def hasNext: Boolean` is still abstract.  That way it can be used without boxing by calling `nextInt` but it is still integrated with Scala collections.